### PR TITLE
Update to node 10

### LINKED ISSUE #2416: Update to node 10
Let's update to node 10.

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #2416.

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
This will drop compatibility with web-ext, as long as web-ext doesn't switch to node 10 we shouldn't do the same, or at least be very certain that we don't have any major updates coming in that web-ext might need (e.g schema updates)

**Comment by EnTeQuAk** [comment]
I think we can attempt to merge this once the Firefox 66 schema update landed.

**Comment by wagnerand** [comment]
I also don't mind waiting for web-ext (if they upgrade sooner, rather than later :) )

File: package.json [hunk 1]
```diff
@@ -6,6 +6,9 @@
   "bin": {
     "addons-linter": "bin/addons-linter"
   },
+  "engines": {
+    "node": ">=10.15"
+  },
```
Comment (EnTeQuAk) [inline package.json]: Let's not enforce this requirement. We are advertising compatibility but as of right now we don't break compatibility with earlier nodejs versions so we shouldn't force users to do so if they dare to use node 8 with the linter.

**Comment by wagnerand** [comment]
I am closing this since web-ext is not moving to node 10 anytime soon.

**Pull request closed without merging**